Filter issues

pcatkins opened the pull request:
Hey guys, this is done now but it should be closely looked at and probably tested before we merge. I've been running this code on our repo and everything seems to be working well. One thing to note is that the labels/keywords that you provide are case sensitive. Meaning if the issue is titled "Deploy ..." you will need to specify "Deploy" as a keyword, "deploy" will not work.

pcatkins committed:
Added command line input for pull request flag and label filter specification

pcatkins committed:
Added filter by labels and pull request flag

pcatkins committed:
Added keyword(s) search of issue title and body

jordan-heemskerk commented:
hey parker, does it fail out for you when you try the following?

![image](https://cloud.githubusercontent.com/assets/4603704/6740210/8b12f270-ce3b-11e4-8177-390d4aa82866.png)

jordan-heemskerk commented on src/main.js:
```diff
@@ -95,7 +124,29 @@ var promptCredentials = function(callback) {
+    //Filter by keyword search 
+    if (argv.keywords) {
+        gKeywords = argv.keywords;
+        gKeywords = gKeywords.split(", ");
```
split by only commas, not on any whitespace. then remove the whitespace from each token. let me know if you dont understand what i mean.

pcatkins replied:
I see what you mean, will do now.

pcatkins commented:
Hmm, I didn't try it on jquery. I suspect it has to do with checking whether issue.body exists and then regardless of the outcome checking issueBody.indexOf(gKeywords[x]) > -1as below.

```
        //Check gKeywords
        if(gKeywords && issue.title !== undefined){
            issueTitle = issue.title;
            if (issue.body !== undefined){
                issueBody = issue.body;
            }   
            for (x in gKeywords) {
                if(issueTitle.indexOf(gKeywords[x]) > -1 || issueBody.indexOf(gKeywords[x]) > -1){
                    if(!counted){
                        if (issue_opened < currentWeek.clone().add(1, "week")  &&  
                                issue_opened > currentWeek) {
                            incrementDataSet(openIssues, weekCounter);
                        }   
                    }   
                }   
            }   
        } 
```

I think title of an issue is a required field. But is body of an issue a required field as well or is this field optional on github?

jordan-heemskerk commented:
ahhh yes i think it may be optional

jordan-heemskerk commented:
can you add a simple check for that, as well as a fix for the whitespace?

pcatkins commented:
You bet, on it now.

jordan-heemskerk commented:
Also your going to have to rebase this onto master before it can merge, that might be a pain, but @gregnr  and i should be able to help

pcatkins committed:
Works for jquery and fixed regex for labels and keywords

pcatkins commented:
Hey @jordan-heemskerk, try it now. I honestly have no idea how I fixed it haha, I added some more logic for counted to make sure an issue can not be counted multiple times and now it runs on jquery. Also, I changed the regex expressions for labels and keywords to only accept strings of the format "bug, defect, ..." so "bug,defect" will throw an error. It's not what you requested but it seemed easier. I can go with your solution if you'd prefer. Let me know.

jordan-heemskerk commented:
Your way is fine. I tested it out and it looks like it works great now. You just need to rebase it. 

To do that:

```
# checkout master branch
git checkout master

# get latest changes to master
git pull

# checkout the FilterIssues branch
git checkout FilterIssues

# replay your changes onto the top of master
git rebase master
```

You'll have some merge conflicts at this point and need to manually resolve them in the files. It can be a pain....

```
# once you have resolved all the conflicts, add the files
git add src/*

# you'll have to do a commit
git commit 

# push up to github
git push origin FilterIssues
```

Once you do that we should be good to merge. I think some of the main things you'll have to do is move some of the stuff that you have in main.js to core.js, due to the changes in 9b4cec1d549151c7f10635e50cadf913ba65cb88.

jordan-heemskerk committed:
Merge branch 'master' into FilterIssues

Conflicts:
	src/analyser.js
	src/main.js

jordan-heemskerk commented:
@pcatkins this is good to merge now I believe

pcatkins commented:
Beauty, thanks @jordan-heemskerk
